### PR TITLE
Simplify tools crafting registration

### PR DESCRIPTION
**Pull request by MoNTE48**
Simplification of the registration code of crafting tools.
52 additions and 218 deletions.
I think that says it all. Adding tools from new materials in the future will be much easier.

### CODE REVIEW

File: mods/default/tools.lua [hunk 1]
```diff
@@ -402,6 +402,58 @@ minetest.register_tool("default:sword_diamond", {
...
+local craft_ingreds = {
+	wood = "group:wood",
+	stone = "group:stone",
+	steel = "default:steel_ingot",
+	bronze = "default:bronze_ingot",
+	mese = "default:mese_crystal",
+	diamond = "default:diamond"
+}
+
+for k, v in pairs(craft_ingreds) do
```
Comment (LoneWolfHT) [inline mods/default/tools.lua]: Maybe change `k` to `name` and `v` to `mat` or `material`?

**Comment by paramat** [comment]
Hmm well ... fewer lines does not mean simplicity. The new code takes more effort to understand and is less readable. Readability has priority over compactness. This is why i usually dislike this kind of change.
However, in this case this is still fairly readable so i will not disapprove, i am neutral.
We do not intend to add new tools from new materials, we already possibly have too many tool tiers.

Just one request: These should probably remain in the crafting file as before, although if other core devs disagree with me on this i do not object.

**Comment by MoNTE48** [comment]
The idea was to have the tools code in the tools.lua file completely.
With the change requested by @LoneWolfHT, it is still easy to read.

**Comment by MoNTE48** [comment]
I can add comments to the tools type that we register, as it was before (Axes, Swords, etc.) Although this is very simple and clear.

**Comment by paramat** [comment]
> The idea was to have the tools code in the tools.lua file

Yes, i quite like doing this, but all other default mod features with dedicated files (torches, chests, furnace, nodes) have crafting recipes in the crafting file. If we move recipes to the dedicated files i think that needs to be discussed and decided first with other core devs.
So this PR may be fine as it is, waiting for input from other core devs on whether to move recipes to dedicated files.

**Comment by LoneWolfHT** [comment]
Code looks fine to me now :+1:

**Comment by paramat** [comment]
SmallJoker seems to support moving the recipes, and i like that too. Will test and probably approve soon.

**Comment by paramat** [comment]
Tested :+1:

**Pull request merged**